Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.CodeAnalysis;
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace Generator
+{
+    [Generator]
+    public class SourceGenerator : ISourceGenerator
+    {
+        private string GetAssemblyName(SourceGeneratorContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyName", out var assemblyName);
+            return assemblyName;
+        }
+
+        private string GetAssemblyVersion(SourceGeneratorContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyVersion", out var assemblyVersion);
+            return assemblyVersion;
+        }
+
+        private string GetWinmdOutputFile(SourceGeneratorContext context)
+        {
+            string winmdDir = Path.Combine(Directory.GetCurrentDirectory(), "Generated Files");

```

Comment:
GeneratedFiles can be defaulted like this, but should also support a project definition (as cppwinrt does)

Reply:
One reason for that is cswinrt can generate different code based on the -tpm switch, in which case we'd want to generated sources under $IntDir

Reply:
Agree on this, going to defer implementing this until we figure out the right place to write to.